src/Orchard.Environment.Extensions/ExtensionLibraryService.cs

```diff
@@ -37,20 +37,19 @@ public class ExtensionLibraryService : IExtensionLibraryService
 
         private readonly Lazy<List<MetadataReference>> _metadataReferences;
         private readonly ApplicationPartManager _applicationPartManager;
-        private readonly IOrchardFileSystem _fileSystem;
+        private readonly ShellOptions _shellOptions;
         private readonly string _probingFolderPath;
         private readonly ILogger _logger;
 
         public ExtensionLibraryService(
             ApplicationPartManager applicationPartManager,
-            IOrchardFileSystem fileSystem,
-            IAppDataFolder appDataFolder,
+            IOptions<ShellOptions> shellOptionsAccessor,
             ILogger<ExtensionLibraryService> logger)
         {
             _metadataReferences = new Lazy<List<MetadataReference>>(GetMetadataReferences);
             _applicationPartManager = applicationPartManager;
-            _fileSystem = fileSystem;
-            _probingFolderPath = appDataFolder.MapPath(ProbingDirectoryName);
+            _shellOptions = shellOptionsAccessor.Value;
+            _probingFolderPath = Path.Combine(_shellOptions.ShellHostContainer.PhysicalPath, ProbingDirectoryName);
```

Comment:
what's the difference between `_shellOptions.ShellHostContainer.PhysicalPath` and `IHostEnvironment.ContentPath` ?